src/parser/parseSVGDocument.ts

```diff
@@ -41,24 +58,20 @@ export function parseSVGDocument(doc, callback, reviver, parsingOptions) {
   }
   parseUseDirectives(doc);
 
-  let svgUid = FabricObject.__uid++,
-    i,
-    len,
-    options = applyViewboxTransform(doc),
-    descendants = Array.from(doc.getElementsByTagName('*'));
+  const svgUid = FabricObject.__uid++,
+    options: ElementsParserParsingOptions = applyViewboxTransform(doc);
+  let descendants = Array.from(doc.getElementsByTagName('*'));
   options.crossOrigin = parsingOptions && parsingOptions.crossOrigin;
   options.svgUid = svgUid;
   options.signal = parsingOptions && parsingOptions.signal;
 
+  // TODO: see about typing isLikelyNode???
+  // @ts-ignore
   if (descendants.length === 0 && isLikelyNode) {
     // we're likely in node, where "o3-xml" library fails to gEBTN("*")
     // https://github.com/ajaxorg/node-o3-xml/issues/21
-    descendants = doc.selectNodes('//*[name(.)!="svg"]');
-    const arr = [];
-    for (i = 0, len = descendants.length; i < len; i++) {
-      arr[i] = descendants[i];
-    }
-    descendants = arr;
+    // TODO: aparently selectNodes is a .NET thing. see about replacing?
+    descendants = (doc as any).selectNodes('//*[name(.)!="svg"]').slice();
```

Comment:
this seems to be some kind of compatibility thing with .NET (internet explorer, I think). Not sure what to do about this. `selectNodes` doesn't exist in the standard lib.

Reply:
is under the `isLikelyNode` so is either a xlmDom parser thing